lib/import/colorectal/providers/leeds/leeds_handler_colorectal.rb

```diff
@@ -1,696 +1,441 @@
+
+              pos_genes -= [gene]
+              varclass = gene_path_hash[gene] || @var_class
+              process_abnormal_gene(rep, genocolorectal, genotypes, gene, varclass)
+            end
+
+            return if pos_genes.blank?
+
+            process_negative_genes(pos_genes, genocolorectal, genotypes)
+          end
+
+          def process_abnormal_gene(rep, genocolorectal, genotypes, gene, varclass)
+            genocolorectal_dup = genocolorectal.dup_colo
+            process_cdna_variant(genocolorectal_dup, rep)
+            process_exonic_variant(genocolorectal_dup, rep)
+            process_protein_impact(genocolorectal_dup, rep)
+            genocolorectal_dup.add_gene_colorectal(gene)
+            genocolorectal_dup.add_variant_class(varclass)
+            @teststatus = 10 if NON_PATH_VARCLASS.include? varclass
+            genocolorectal_dup.add_status(@teststatus)
+            genotypes.append(genocolorectal_dup)
+          end
+
+          # For all negative genes add a genotype with status 1
+          def process_negative_genes(negative_genes, genocolorectal, genotypes)
+            negative_genes&.each do |gene|
+              genocolorectal_dup = genocolorectal.dup_colo
+              genocolorectal_dup.add_gene_colorectal(gene)
+              genocolorectal_dup.add_status(1)
+              genotypes << genocolorectal_dup
+            end
+          end
+
+          # Prepare genes panel with priority of report over genotype
+          def allocate_genes
+            @genes_panel = get_genes_from_report(@report)
+
+            return if @genes_panel.present?
+
+            GENES_PANEL.each do |panel, genes|
+              @genes_panel = genes if @genes_hash[panel].include?(@geno)
             end
           end
 
-          def process_scope(geno, genocolorectal, record)
-            scope = Maybe(record.raw_fields['reason']).
-                    or_else(Maybe(record.mapped_fields['genetictestscope']).or_else(''))
-            # ------------ Set the test scope ---------------------
-            if (geno.downcase.include? 'ashkenazi') || (geno.include? 'AJ')
-              genocolorectal.add_test_scope(:aj_screen)
+          # Get test status for record
+          def find_test_status(_record)
+            @teststatus = nil
+            if @geno == 'ngs msh2 seq variant'
+              @teststatus = @report.include?('likely to be benign') ? 10 : 2
             else
-              stripped_scope = TEST_SCOPE_MAP_COLO[scope.downcase.strip]
-              genocolorectal.add_test_scope(stripped_scope) if stripped_scope
+              STATUS_PANEL.each do |category, status|
+                @teststatus = status if @status_hash[category].include?(@geno)
+              end
+            end
+            # exceptional cases for 'mlh1 only -ve' genotype
+            exceptinal_teststatus if @geno == 'mlh1 only -ve'
+            @teststatus
+          end
+
+          def exceptinal_teststatus
```

Comment:
Should this be `exceptional_teststatus`?

Reply:
Thanks @ollietulloch , done in latest commit .